docs/docs/ecommerce-tutorial/index.md

```diff
@@ -0,0 +1,542 @@
+
+```shell
+npm install gatsby-plugin-stripe-checkout
+```
+
+Open the root site directory in a text editor and navigate to `gatsby-config.js` and add stripe-checkout plugin to `gatsby-config.js` in the plugins section. Your `gatsby-config.js` should look like the following code example:
+
+```jsx
+module.exports = {
+ siteMetadata: {
+   title: 'Gatsby Default Starter',
+ },
+ plugins: ['gatsby-plugin-react-helmet', 'gatsby-plugin-stripe-checkout'],
+}
+```
+
+
+## See your site hot reload in the browser!
+
+Run `gatsby develop` in the terminal, which starts a development server and reloads changes you make to your site so you can preview them in the browser (you can turn off the server by pressing Ctrl + C). Open up your browser to localhost:8000 and you should see a default homepage.
+
+> **NOTE**: If you have already started your gatsby development server using `gatsby develop` you will need to restart the server by pressing CTRL + C and running the command again to see changes in your gatsby.config reflected on localhost***
+
+
+## How does the plugin work?
+
+Stripe Checkout processes payments with information we send it, you can read more about how it works in Stripe’s docs. The Gatsby plugin, `gatsby-plugin-stripe-checkout`, will add this snippet:
+
+```shell
+<script src="https://checkout.stripe.com/checkout.js"></script>
+```
+
+to the end of the `<body>` tag across all of your pages in your Gatsby project, allowing you to call the Stripe checkout methods from Stripe’s API to create charges.
+
+If you’d like more extensive access to Stripe data, explore more Gatsby / Stripe plugins.
+
+
+## Creating a button
+
+There is a default checkout modal that is available through the plugin we’re using, shown in the image below. When we call the `.open()` method through Stripe Checkout, the rest of the screen is darkened and a modal appears over the top, directing the user’s attention to the checkout form. You’ll need to create a button component that calls this method or triggers Stripe from your site.
+
+> **NOTE**: If you’d like to customize the checkout modal, see the [Stripe Checkout Elements plugin](/packages/gatsby-plugin-stripe-elements/?=stripe)
+
+![Stripe card payments modal](card-payments-modal.png)
+
+> Notice the input fields for email and credit card information—this design is already made by Stripe for you to use without styling anything. 
+
+In order to launch the Stripe Checkout modal (shown above), you need to call the `.configure()` method from Stripe Checkout in your Gatsby site. The only information you are required to provide is your publishable API key so that Stripe knows what account on their platform to send payments to. You can read about other recommended and optional configurations like currency or shipping information in the Stripe Checkout Reference. You are going to make a `checkout.js` file that handles this configuration.
```

Comment:
I'm confused; the paragraph above says we need to call `.open()` method and this one says `.configure()`. Which one is it?

Reply:
Hmm.. It looks like both. `configure()` sets up Stripe and I think it automatically runs every time the page loads. We run `open()` when the buy button is clicked, which then triggers the Stripe payment overlay to open.

Maybe we could add a section before the _Creating a button_ heading that describes this sequence of events? 

- _page load_ -> `configure()`
- _some time later..._
- _user clicks button_ -> `open()`